src/d3a/models/area/__init__.py

```diff
@@ -427,14 +428,18 @@ def _match_bids_offers(self) -> None:
                 bid_offer_pairs = bid_offer_matcher.get_matches_recommendations(data)
                 if not bid_offer_pairs:
                     break
-                if hasattr(bid_offer_matcher.match_algorithm, "state"):
-                    self.state.cumulative_offers[self.now] = (bid_offer_matcher.match_algorithm.
-                                                              state.cumulative_offers)
-                    self.state.cumulative_bids[self.now] = (bid_offer_matcher.match_algorithm.
-                                                            state.cumulative_bids)
-
+                self.copy_clearing_state_to_area_state()
                 market.match_recommendations(bid_offer_pairs)
 
+    def copy_clearing_state_to_area_state(self):
+        """Keeping copy of myco matcher's cumulative_offers/cumulative_bids to area for
+        supply/demand plot"""
+        if hasattr(bid_offer_matcher.match_algorithm, "state"):
+            self.state.cumulative_offers[self.now] = (bid_offer_matcher.match_algorithm.
+                                                      state.cumulative_offers)
+            self.state.cumulative_bids[self.now] = (bid_offer_matcher.match_algorithm.
```

Comment:
Why are you storing the whole `bid_offer_matcher.match_algorithm.state.cumulative_offers` in the `self.state.cumulative_offers` only in the `self.now` key ?

Reply:
Because that is needed to plot supply/demand curve for area at that point of time.